system/Validation/Validation.php

```diff
@@ -499,7 +500,7 @@ public function hasRule(string $field): bool
      *
      * @param string $group Group.
      *
-     * @throws InvalidArgumentException If group not found.
+     * @throws ValidationException If group not found.
```

Comment:
`InvalidArgumentException` seems like the correct throw here. It doesn't look like you touched the actual method - was this some previous change to consolidate exceptions or something?

Reply:
It seems it was InvalidArgumentException but changed to ValidationException.
I just update the doc comment.
See the code: https://github.com/codeigniter4/CodeIgniter4/blob/a127d7190dcb709488d53104c8c84993b8233bd0/system/Validation/Validation.php#L507-L518

Reply:
Beyond the scope of this PR but I think that is a mistake. Catching a "validation exception" should not cover logic errors that were unrelated to the validation process itself.

Reply:
Never mind, I have it backwards: exceptions are never thrown when input fails validation rules. These are correct.